fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingCommon.java

```diff
@@ -287,8 +287,5 @@ public boolean loadConfig() {
 
     public void close() {
         runner.close();
-        if (synchronizedSessionRunner != null) {
```

Comment:
Another option would be to _only_ close the `synchronizedSessionRunner` if it is non-null, and only close the `runner` if `synchronizedSessionRunner` is null.

Reply:
Since we do not control the timing of the other thread, checking if the `syncRunner` is null may cause some weird race conditions. So I thought that it would be safer to only close to `mainRunner`, and only in this path. 
Although it it does is to close the underlying runner, it does feel weird that the `syncRunner` is never being closed. I hope to replace this mechanism ASAP.